spec/ics-3-connection-semantics/opening_handshake.tex

```diff
@@ -0,0 +1,82 @@
+\include{common}
+
+\begin{document}
+\begin{varwidth}{\linewidth}
+
+\begin{algorithmic}
+\Struct[Datagram]{CONNOPENINIT}
+\Field{Identifier}{identifier}
+\Field{Identifier}{desiredCounterpartyIdentifier}
+\Field{Version}{desiredVersion}
+\Field{RootOfTrust}{rootOfTrust}
+\EndStruct
+\end{algorithmic}
+
+\begin{algorithmic}
+\Struct[Datagram]{CONNOPENTRY}
+\Field{Identifier}{desiredIdentifier}
+\Field{Identifier}{counterpartyIdentifier}
+\Field{Version}{desiredVersion}
+\Field{RootOfTrust}{rootOfTrust}
+\Field{AccumulatorProof}{proofInit}
+\EndStruct
+\end{algorithmic}
+
+\begin{algorithmic}
+\Struct[Datagram]{CONNOPENACK}
+\Field{Identifier}{identifier}
+\Field{Identifier}{agreedCounterpartyIdentifier}
+\Field{Version}{agreedVersion}
+\Field{AccumulatorProof}{proofTry}
+\EndStruct
+\end{algorithmic}
+
+\begin{algorithmic}
+\Struct[Datagram]{CONNOPENCONFIRM}
+\Field{Identifier}{identifier}
+\Field{AccumulatorProof}{proofAck}
+\EndStruct
+\end{algorithmic}
+
+\begin{algorithmic}
+\Function{handleDatagram}{datagram}
+  \Switch{$datagram$}
+    \Case{$CONNOPENINIT$}
+      \State $state \gets INIT$
+      \State \Call{Set}{identifier, (state, desiredVersion, desiredCounterpartyIdentifier, rootOfTrust)}
```

Comment:
Does this need to check there is no open, init, or any (connection) state known?

Reply:
Appropriately permissioned to who called `ConnOpenInit`, but that seems like a sensible default.

Updated to assert that no current connection exists.